types/i18next/index.d.ts

```diff
@@ -1,157 +1,711 @@
+
+        /**
+         * language to use if translations in user language are not available.
+         * @default dev
+         */
+        fallbackLng?: false | FallbackLng;
+
+        /**
+         * 	array of allowed languages
+         * @default false
+         */
+        whitelist?: false | string[];
+
+        /**
+         * if true will pass eg. en-US if finding en in whitelist
+         * @default false
+         */
+        nonExplicitWhitelist?: boolean;
+
+        /**
+         * language codes to lookup, given set language is
+         * 'en-US': 'all' --> ['en-US', 'en', 'dev'],
+         * 'currentOnly' --> 'en-US',
+         * 'languageOnly' --> 'en'
+         * @default all
+         */
+        load?: "all" | "currentOnly" | "languageOnly";
+
+        /**
+         * array of languages to preload. Important on serverside to assert translations are loaded before rendering views.
+         * @default false
+         */
+        preload?: false | string[];
+
+        /**
+         * language will be lowercased eg. en-US --> en-us
+         * @default false
+         */
+        lowerCaseLng?: boolean;
+
+        /**
+         * string or array of namespaces to load
+         * @default translation
+         */
         ns?: string | string[];
+
+        /**
+         * files to load
+         * @see https://www.i18next.com/principles/fallback.html#namespace-fallback
+         */
+        namespaces?: string | string[];
+
+        /**
+         * default namespace used if not passed to translation function
+         * @default translation
+         */
         defaultNS?: string;
-        fallbackNS?: string | string[];
-        whitelist?: string[];
-        lowerCaseLng?: boolean;
-        load?: string
-        preload?: string[];
-        keySeparator?: string | false;
-        nsSeparator?: string | false;
-        pluralSeparator?: string;
-        contextSeparator?: string;
+
+        /**
+         * string or array of namespaces to lookup key if not found in given namespace.
+         * @default false
+         */
+        fallbackNS?: false | string | string[];
+
+        /**
+         * calls save missing key function on backend if key not found
+         * @default false
+         */
         saveMissing?: boolean;
-        saveMissingTo?: string;
-        missingKeyHandler?: (lng: string, ns: string, key: string, fallbackValue: string) => void;
-        parseMissingKeyHandler?: (key: string) => void;
+
+        /**
+         * 'current' or 'all'
+         * @default fallback
+         */
+        saveMissingTo?: "current" | "all" | "fallback";
+
+        /**
+         * function(lng, ns, key, fallbackValue) { } used for custom missing key handling (needs saveMissing set to true!)
+         * @default false
+         */
+        missingKeyHandler?: false | ((lng: string, ns: string, key: string, fallbackValue: string) => void);
+
+        /**
+         * function(key) { // return value to display }
```

Comment:
Please look at [the PR](https://github.com/i18next/i18next-gitbook/pull/9) and update comment.